internal/resource/config/token_processor_to_token_generator_mapping_resource.go

```diff
@@ -0,0 +1,334 @@
+				Description: "The id of the Token Processor.",
+				Required:    true,
+			},
+			"issuance_criteria": issuancecriteria.IssuanceCriteriaSchema(),
+		},
+	}
+	AddCommonSchema(&schema)
+	resp.Schema = schema
+}
+
+func addOptionalTokenProcessorToTokenGeneratorMappingFields(ctx context.Context, addRequest *client.TokenToTokenMapping, plan tokenProcessorToTokenGeneratorMappingsResourceModel) error {
+	var err error
+	if internaltypes.IsDefined(plan.AttributeSources) {
+		attributeSourcesAttr := plan.AttributeSources.Elements()
+		addRequest.AttributeSources = []client.AttributeSourceAggregation{}
+		for _, source := range attributeSourcesAttr {
+			//Determine which attribute source type this is
+			sourceAttrs := source.(types.Object).Attributes()
+			attributeSourceInner := client.AttributeSourceAggregation{}
+			if internaltypes.IsDefined(sourceAttrs["custom_attribute_source"]) {
+				attributeSourceInner.CustomAttributeSource = &client.CustomAttributeSource{}
+				err = json.Unmarshal([]byte(internaljson.FromValue(sourceAttrs["custom_attribute_source"], true)), attributeSourceInner.CustomAttributeSource)
+			}
+			if internaltypes.IsDefined(sourceAttrs["jdbc_attribute_source"]) {
+				attributeSourceInner.JdbcAttributeSource = &client.JdbcAttributeSource{}
+				err = json.Unmarshal([]byte(internaljson.FromValue(sourceAttrs["jdbc_attribute_source"], true)), attributeSourceInner.JdbcAttributeSource)
+			}
+			if internaltypes.IsDefined(sourceAttrs["ldap_attribute_source"]) {
+				attributeSourceInner.LdapAttributeSource = &client.LdapAttributeSource{}
+				err = json.Unmarshal([]byte(internaljson.FromValue(sourceAttrs["ldap_attribute_source"], true)), attributeSourceInner.LdapAttributeSource)
+			}
+			if err != nil {
+				return err
+			}
+			addRequest.AttributeSources = append(addRequest.AttributeSources, attributeSourceInner)
+		}
+	}
+
+	if internaltypes.IsDefined(plan.IssuanceCriteria) {
+		addRequest.IssuanceCriteria = client.NewIssuanceCriteria()
+		if internaltypes.IsDefined(plan.IssuanceCriteria.Attributes()["conditional_criteria"]) {
+			addRequest.IssuanceCriteria.ConditionalCriteria = []client.ConditionalIssuanceCriteriaEntry{}
+			conditionalCriteriaErr := json.Unmarshal([]byte(internaljson.FromValue(plan.IssuanceCriteria.Attributes()["conditional_criteria"].(types.List), true)), &addRequest.IssuanceCriteria.ConditionalCriteria)
+			if conditionalCriteriaErr != nil {
+				return conditionalCriteriaErr
+			}
+		}
+		if internaltypes.IsDefined(plan.IssuanceCriteria.Attributes()["expression_criteria"]) {
+			addRequest.IssuanceCriteria.ExpressionCriteria = []client.ExpressionIssuanceCriteriaEntry{}
+			expressionCriteriaErr := json.Unmarshal([]byte(internaljson.FromValue(plan.IssuanceCriteria.Attributes()["expression_criteria"].(types.List), true)), &addRequest.IssuanceCriteria.ExpressionCriteria)
+			if expressionCriteriaErr != nil {
+				return expressionCriteriaErr
+			}
+		}
+	}
+
+	if internaltypes.IsDefined(plan.Id) {
```

Comment:
This should use customid. Id is readonly

Reply:
changed to CustomId